pdata/internal/metrics.go

```diff
@@ -192,19 +192,21 @@ const (
 	MetricDataPointFlagsNone = MetricDataPointFlags(otlpmetrics.DataPointFlags_FLAG_NONE)
 )
 
-// NewMetricDataPointFlags returns a new MetricDataPointFlags combining the flags passed
-// in as parameters.
-func NewMetricDataPointFlags(flags ...MetricDataPointFlag) MetricDataPointFlags {
-	var flag MetricDataPointFlags
-	for _, f := range flags {
-		flag |= MetricDataPointFlags(f)
-	}
-	return flag
+// NewMetricDataPointFlags returns a new MetricDataPointFlags with no flags set.
+func NewMetricDataPointFlags() MetricDataPointFlags {
```

Comment:
You need one more change, since this type is no longer immutable, we need to remove the setter from the parent struct

Reply:
The reason is because that is the pattern we use

Reply:
I am not sure what you mean by "parent struct"

Reply:
https://github.com/open-telemetry/opentelemetry-collector/blob/c368d497a2046011da4a39ee977ceed5f841c266/pdata/internal/generated_pmetric.go#L1209